### PR TITLE
[IMP] calendar: Improve color and filters

### PR DESCRIPTION
**Pull request by jbm-odoo**
Now in a calendar view, if we use the color, attribute, it will use
the same color as Gantt views and the color picker widget.

The attribute color isn't more set as default in filter because it make,
in most situation, no sense to filter by color.

Now, we can add new filters in the right panel. To do that, we must
add the attribute "filters='1'" in the corresponding field line.
And if the filter has no direct link with the color of the model,
we can specify an attribute 'color' for this filter line block.
(for example color='color', and the color of the related model
will be used)

TaskID: 2153249

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/views/calendar/calendar_model.js [hunk 1]
```diff
@@ -503,7 +503,7 @@ return AbstractModel.extend({
                 var value = event.record[fieldName];
                 event.color_index = _.isArray(value) ? value[0] : value;
             });
-            this.model_color = this.fields[fieldName].relation || element.model;
+            this.model_color = this.fields[fieldName].relation || this.modelName;
```
Comment (Polymorphe57) [inline addons/web/static/src/js/views/calendar/calendar_model.js]: Why this change? Is it a fix?

File: addons/web/static/src/js/views/calendar/calendar_model.js [hunk 2]
```diff
@@ -651,6 +671,13 @@ return AbstractModel.extend({
                         f.label = to_read[f.avatar_model][f.value];
                     });
                 }
+                if (filter.color_model && filter.field_color) {
+                    _.each(filter.filters, function (f) {
+                        if (!f.color_index) {
+                            f.color_index = color_filter[filter.color_model] && color_filter[filter.color_model][f.value];
```
Comment (Polymorphe57) [inline addons/web/static/src/js/views/calendar/calendar_model.js]: I would do that earlier. I might be wrong but it seems there is no need to define an object color_filter.

File: addons/calendar/views/calendar_views.xml [hunk 3]
```diff
@@ -263,7 +263,7 @@
                 event_limit="5"
                 color="partner_id">
                 <field name="attendee_status"/>
-                <field name="partner_id"/>
+                <field name="partner_id" filter="1"/>
```
Comment (Polymorphe57) [inline addons/calendar/views/calendar_views.xml]: New attribute implies a doc update :-) (view.rst)

File: addons/web/static/src/js/views/calendar/calendar_view.js [hunk 4]
```diff
@@ -92,25 +92,32 @@ var CalendarView = AbstractView.extend({
...
                 }
+                if (child.attrs.filter) {
+                    filters[fieldName] = filters[fieldName] || {
+                        'title': fields[fieldName].string,
+                        'fieldName': fieldName,
+                        'filters': [],
+                    };
+                    if (child.attrs.color) {
+                        filters[fieldName].field_color = child.attrs.color;
+                        filters[fieldName].color_model = fields[fieldName].relation;
```
[the remark this answers is not in the document]
Comment (jbm-odoo) [reply addons/web/static/src/js/views/calendar/calendar_view.js]: No it isn't always defined, but it doesn't matter, it can be undefined

**Comment by jbm-odoo** [comment]
@Polymorphe57 Thanks for your review

**Comment by tivisse** [comment]
@robodoo r+ rebase-ff
